src/leap/mail/imap/mailbox.py

```diff
@@ -104,65 +103,31 @@ class SoledadMailbox(WithMsgFields, MBoxParser):
     CMD_UIDVALIDITY = "UIDVALIDITY"
     CMD_UNSEEN = "UNSEEN"
 
-    # FIXME we should turn this into a datastructure with limited capacity
+    # TODO we should turn this into a datastructure with limited capacity
     _listeners = defaultdict(set)
 
-    next_uid_lock = threading.Lock()
-    last_uid_lock = threading.Lock()
-
-    # TODO unify all the `primed` dicts
-    _fdoc_primed = {}
-    _last_uid_primed = {}
-    _known_uids_primed = {}
-
-    def __init__(self, mbox, soledad, memstore, rw=1):
+    def __init__(self, collection, rw=1):
         """
-        SoledadMailbox constructor. Needs to get passed a name, plus a
-        Soledad instance.
-
-        :param mbox: the mailbox name
-        :type mbox: str
+        SoledadMailbox constructor.
 
-        :param soledad: a Soledad instance.
-        :type soledad: Soledad
-
-        :param memstore: a MemoryStore instance
-        :type memstore: MemoryStore
+        :param collection: instance of IMAPMessageCollection
+        :type collection: IMAPMessageCollection
 
         :param rw: read-and-write flag for this mailbox
         :type rw: int
         """
-        leap_assert(mbox, "Need a mailbox name to initialize")
-        leap_assert(soledad, "Need a soledad instance to initialize")
-
-        from twisted.internet import reactor
-        self.reactor = reactor
-
-        self.mbox = self._parse_mailbox_name(mbox)
         self.rw = rw
-
-        self._soledad = soledad
-        self._memstore = memstore
-
-        self.messages = MessageCollection(
-            mbox=mbox, soledad=self._soledad, memstore=self._memstore)
+        self.closed = False
```

Comment:
Hum.. I confused the use of "closed" on account and on mailbox.
I've cleaned up the use of the account session termination, and removed several cruft in mailbox. I think I can remove all the close attributes since I don't implement that interface anymore, but I'd like to check that carefully.